hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -182,15 +192,23 @@ long uploadJobResources(JobConfig jobConfig) {
         try {
             for (ResourceConfig rc : jobConfig.getResourceConfigs()) {
                 switch (rc.getResourceType()) {
-                    case REGULAR_FILE:
-                        InputStream in = rc.getUrl().openStream();
-                        readStreamAndPutCompressedToMap(rc.getId(), tmpMap, in);
-                        in.close();
+                    case CLASS:
+                        try (InputStream in = rc.getUrl().openStream()) {
+                            readStreamAndPutCompressedToMap(rc.getId(), tmpMap, in);
+                        }
+                        break;
+                    case FILE:
+                        IMapOutputStream os = new IMapOutputStream(getJobFileStorage(jobId).get(), rc.getId());
+                        zipFileToOutputStream(Paths.get(rc.getUrl().getFile()), os);
```

Comment:
What if the URL isn't a local file? It will break. We should use `rc.getUrl().openStream()`. 

Also the `zipFileToOutputStream` is named incorrectly. What it does is that it puts a file as a sole entry in a zip file.

Reply:
It's worse, it actually fails on Windows with:
```
java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/work/hazelcast/hazelcast-jet/hazelcast-jet-core/target/test-classes/deployment/resource.txt

	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at com.hazelcast.jet.impl.JobRepository.uploadJobResources(JobRepository.java:202)
	at com.hazelcast.jet.impl.AbstractJetInstance.uploadResourcesAndAssignId(AbstractJetInstance.java:149)
	at com.hazelcast.jet.impl.AbstractJetInstance.newJob(AbstractJetInstance.java:57)
```
We must use `URL.openStream()`.